### PR TITLE
Add ppc64le platform support for build-cross.sh script

### PR DESCRIPTION
**Pull request by mkumatag**
This PR will make sure that build-cross.sh script works on power64le platform.

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@smarterclayton any reason why we couldn't target `ppc64le`?

**Comment by smarterclayton** [comment]
We would need to double check that the release binaries will work.  Post 1.3 I will turn this on and test with it.  I don't see any reason not to test it.

**Comment by mkumatag** [comment]
@smarterclayton I'm working on making this build to work on ppc64le platform as well, soon will propose changes to make it work on ppc64le platform as well including release binaries.

**Comment by stevekuznetsov** [comment]
> soon will propose changes to make it work on ppc64le platform as well including release binaries.

@mkumatag If you can, bring those changes into this PR so we can merge them together.

**Comment by mkumatag** [comment]
@stevekuznetsov submitted a patch to make build-cross.sh on ppc64le platform, can you please review.

**Comment by mkumatag** [comment]
Here is the list output of this script on ppc64le machine:

```
[root@rhel72 origin]# ./hack/build-cross.sh 
++ Building go targets for linux/amd64: cmd/openshift cmd/oc
++ Building go targets for darwin/amd64: cmd/openshift cmd/oc
++ Building go targets for windows/amd64: cmd/openshift cmd/oc
++ Building go targets for linux/386: cmd/openshift cmd/oc
++ Building go targets for linux/ppc64le: cmd/openshift cmd/oc
++ Building go targets for linux/ppc64le: images/pod cmd/dockerregistry cmd/gitserver examples/hello-openshift examples/deployment
++ Placing binaries
++ Creating openshift-origin-client-tools-v1.3.0-alpha.1+b9ec174-dirty-b9ec174-linux-64bit.tar.gz
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./openshift
70e60f9df2799f1464b66d503f0cefbce670794058d6d55daf3264d98c40a367  ./oc
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./oadm
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./kubelet
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./kube-proxy
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./kube-apiserver
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./kube-controller-manager
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./kube-scheduler
70e60f9df2799f1464b66d503f0cefbce670794058d6d55daf3264d98c40a367  ./kubectl
abbb6ae590e75b19affa8391c4c4b7d4d118f6a39688860a41784a5d7e545db0  ./README.md
f6d5b461deb8038ce0e083c9cb7f59859caa04c9b4f72149367393e9b252cf14  ./LICENSE
++ Creating openshift-origin-server-v1.3.0-alpha.1+b9ec174-dirty-b9ec174-linux-64bit.tar.gz
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./openshift
70e60f9df2799f1464b66d503f0cefbce670794058d6d55daf3264d98c40a367  ./oc
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./oadm
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./kubelet
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./kube-proxy
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./kube-apiserver
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./kube-controller-manager
9d9018851009a23979a6a8678d943426d9430885726cd47c915be1d1cb176073  ./kube-scheduler
70e60f9df2799f1464b66d503f0cefbce670794058d6d55daf3264d98c40a367  ./kubectl
abbb6ae590e75b19affa8391c4c4b7d4d118f6a39688860a41784a5d7e545db0  ./README.md
f6d5b461deb8038ce0e083c9cb7f59859caa04c9b4f72149367393e9b252cf14  ./LICENSE
++ Creating openshift-origin-client-tools-v1.3.0-alpha.1+b9ec174-dirty-b9ec174-mac.zip
47e69e28b4e4987e1c1c9ef9c65ca00afc8602196d63c6d719359a0bf5b11e9d  oc
abbb6ae590e75b19affa8391c4c4b7d4d118f6a39688860a41784a5d7e545db0  README.md
f6d5b461deb8038ce0e083c9cb7f59859caa04c9b4f72149367393e9b252cf14  ./LICENSE
++ Creating openshift-origin-client-tools-v1.3.0-alpha.1+b9ec174-dirty-b9ec174-windows.zip
28712ff9790c75444eaa07edf716a4544e814e4b2f022af0a6f6418f7d1f7811  oc.exe
abbb6ae590e75b19affa8391c4c4b7d4d118f6a39688860a41784a5d7e545db0  README.md
f6d5b461deb8038ce0e083c9cb7f59859caa04c9b4f72149367393e9b252cf14  ./LICENSE
++ Creating openshift-origin-client-tools-v1.3.0-alpha.1+b9ec174-dirty-b9ec174-linux-32bit.tar.gz
980d6442df90f2cb51eee54e5823c28f153f87b7c34da68b967796dcd3f1a5d8  ./openshift
485fec773c35c7344b90de057c5f5a9c05ea6e5e84a2fefcd1b2205bafbc86b4  ./oc
980d6442df90f2cb51eee54e5823c28f153f87b7c34da68b967796dcd3f1a5d8  ./oadm
980d6442df90f2cb51eee54e5823c28f153f87b7c34da68b967796dcd3f1a5d8  ./kubelet
980d6442df90f2cb51eee54e5823c28f153f87b7c34da68b967796dcd3f1a5d8  ./kube-proxy
980d6442df90f2cb51eee54e5823c28f153f87b7c34da68b967796dcd3f1a5d8  ./kube-apiserver
980d6442df90f2cb51eee54e5823c28f153f87b7c34da68b967796dcd3f1a5d8  ./kube-controller-manager
980d6442df90f2cb51eee54e5823c28f153f87b7c34da68b967796dcd3f1a5d8  ./kube-scheduler
485fec773c35c7344b90de057c5f5a9c05ea6e5e84a2fefcd1b2205bafbc86b4  ./kubectl
abbb6ae590e75b19affa8391c4c4b7d4d118f6a39688860a41784a5d7e545db0  ./README.md
f6d5b461deb8038ce0e083c9cb7f59859caa04c9b4f72149367393e9b252cf14  ./LICENSE
++ Creating openshift-origin-client-tools-v1.3.0-alpha.1+b9ec174-dirty-b9ec174-linux-ppc64le.tar.gz
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./openshift
64ef3d9931b01384da4ba5ee11f3637ffa1fb8d6b3c7ac706bdab68632e89484  ./oc
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./oadm
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./kubelet
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./kube-proxy
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./kube-apiserver
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./kube-controller-manager
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./kube-scheduler
64ef3d9931b01384da4ba5ee11f3637ffa1fb8d6b3c7ac706bdab68632e89484  ./kubectl
abbb6ae590e75b19affa8391c4c4b7d4d118f6a39688860a41784a5d7e545db0  ./README.md
f6d5b461deb8038ce0e083c9cb7f59859caa04c9b4f72149367393e9b252cf14  ./LICENSE
++ Creating openshift-origin-server-v1.3.0-alpha.1+b9ec174-dirty-b9ec174-linux-ppc64le.tar.gz
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./openshift
64ef3d9931b01384da4ba5ee11f3637ffa1fb8d6b3c7ac706bdab68632e89484  ./oc
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./oadm
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./kubelet
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./kube-proxy
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./kube-apiserver
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./kube-controller-manager
a77c2cef39deedcf6c9b6f292b01d90a0365922b3823b53497c8dbb5c4044b35  ./kube-scheduler
64ef3d9931b01384da4ba5ee11f3637ffa1fb8d6b3c7ac706bdab68632e89484  ./kubectl
abbb6ae590e75b19affa8391c4c4b7d4d118f6a39688860a41784a5d7e545db0  ./README.md
f6d5b461deb8038ce0e083c9cb7f59859caa04c9b4f72149367393e9b252cf14  ./LICENSE
++ Placing binaries
++ Creating openshift-origin-image-v1.3.0-alpha.1+b9ec174-dirty-b9ec174-linux-ppc64le.tar.gz
baa5c7ec11883870cac0d4f50a1ab4f87d13cabcc2a03e7db60b84da5f53a3d4  ./hello-openshift
c466b4cbb8e732adfc8258a147d1320012a76dc2c3427d98809527d7915a97e2  ./deployment
50b003b725103ba8e9392bd57017ad7dbec39e3f5ba3a7bd978c070814d4723d  ./pod
3b936cbe329e3cc20826ab6bd73ecd603227cda6dc1dd2bd314f289782eb05b1  ./dockerregistry
92184fbf2042b56e47b16090aeee10e30dcef23a6f7e43cd8b87a9d57c6ede77  ./gitserver
./hack/build-cross.sh took 987 seconds
[root@rhel72 origin]# 
```

File: hack/common.sh [hunk 1]
```diff
@@ -29,12 +35,20 @@ readonly OS_SCRATCH_IMAGE_COMPILE_TARGETS=(
...
 readonly OS_IMAGE_COMPILE_BINARIES=("${OS_SCRATCH_IMAGE_COMPILE_TARGETS[@]##*/}" "${OS_IMAGE_COMPILE_TARGETS[@]##*/}")
 
-readonly OS_CROSS_COMPILE_PLATFORMS=(
+OS_CROSS_COMPILE_PLATFORMS=(
   linux/amd64
   darwin/amd64
   windows/amd64
   linux/386
 )
+if [[ "$(os::build::host_platform)" == "linux/ppc64le" ]]; then
```
Comment (stevekuznetsov) [inline hack/common.sh]: Why can we only target `ppc64le` if we have it as a host?

Comment (mkumatag) [inline hack/common.sh]: Currently there is known issue with golang and it is failing to link bigger binaries. So I'm intentionally avoiding to consider ppc64le for cross compiling other than on ppc64le host platform.

Currently I'm running with a patched golang to build openshift or even I can use gccgo to build. Once fix is available in the golang community, I'll add `linux/ppc64` to all host platforms.

Comment (stevekuznetsov) [inline hack/common.sh]: Unless the issues actually make it impossible to do this cross-compilation on platforms other than `ppc64le`, we should just keep the list of targets full and static -- having this added on certain hosts kind of breaks the whole point of `hack/build-cross.sh`.

Comment (mkumatag) [inline hack/common.sh]: I completely agree. Time being we don't have any other option, I'll add ppc64le platform to target list as soon as I can when I'm able to build on amd64 platform with golang without any issues.

**Comment by mkumatag** [comment]
@stevekuznetsov @smarterclayton let me know if any other comments.

File: hack/common.sh (hunk 1 above)
Comment (mkumatag) [inline hack/common.sh]: Here is the link of pending PR for golang - https://go-review.googlesource.com/#/c/27790

**Comment by mkumatag** [comment]
As openshift origin 1.3 is already out, @smarterclayton can you please have a look into this PR?

**Comment by smarterclayton** [comment]
Yup, sorry for the delay.  [test]

**Comment by mkumatag** [comment]
@smarterclayton seems like vagrant failed failed to bring up the test env, can you please tests once again?

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
LGTM [merge], thanks

**Pull request merged**